### PR TITLE
Provide metrics of k8s API execution

### PR DESCRIPTION
**Pull request by skabashnyuk**
# What does this PR do?
Provide metrics of k8s API execution
Знімок екрана  о 20 09 14

### What issues does this PR fix or reference?
n/a

#### Release Notes
n/a

#### Docs PR
TODO

### CODE REVIEW

**Comment by skabashnyuk** [comment]
CC @ibuziuk

**Review by sleshchenko — APPROVED** [review]
:+1: Should be really useful to understand how fast cluster is.

**Review by amisevsk — APPROVED** [review]
Should be very useful

File: core/che-core-metrics-core/src/main/java/org/eclipse/che/core/metrics/NopMetricsModule.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.core.metrics;
+
+import com.google.inject.AbstractModule;
+import okhttp3.EventListener;
+
+public class NopMetricsModule extends AbstractModule {
```
Comment (sparkoo) [inline core/che-core-metrics-core/src/main/java/org/eclipse/che/core/metrics/NopMetricsModule.java]: I like `Noop` more than `Nop` and it's more frequent in the codebase

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java [hunk 2]
```diff
@@ -61,15 +62,16 @@ public KubernetesClientFactory(
           int maxConcurrentRequestsPerHost,
       @Named("che.infra.kubernetes.client.http.connection_pool.max_idle") int maxIdleConnections,
       @Named("che.infra.kubernetes.client.http.connection_pool.keep_alive_min")
-          int connectionPoolKeepAlive) {
+          int connectionPoolKeepAlive,
+      EventListener eventListener) {
```
Comment (sparkoo) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java:66]: formatting looks a bit off here

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java [hunk 3]
```diff
@@ -66,14 +67,16 @@ public OpenShiftClientFactory(
           int maxConcurrentRequestsPerHost,
       @Named("che.infra.kubernetes.client.http.connection_pool.max_idle") int maxIdleConnections,
       @Named("che.infra.kubernetes.client.http.connection_pool.keep_alive_min")
-          int connectionPoolKeepAlive) {
+          int connectionPoolKeepAlive,
+      EventListener eventListener) {
```
Comment (sparkoo) [inline infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java:71]: formatting looks a bit off here

**Review by sparkoo — APPROVED** [review]
nice. just few nitpicks

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java (hunk 2 above)
Comment (skabashnyuk) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java:66]: My offer to the god of formatting wasn't accepted. I think that is the reason why it looks like that and we have no way not to accept it. I'm sorry.

File: infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java (hunk 3 above)
Comment (skabashnyuk) [reply infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java:71]: My offer to the god of formatting wasn't accepted. I think that is the reason why it looks like that and we have no way not to accept it. I'm sorry.

**Comment by skabashnyuk** [comment]
ci-test

**Comment by skabashnyuk** [comment]
crw-ci-test

**Comment by skabashnyuk** [comment]
crw-ci-test

**Pull request merged**